interfaces/builtin/desktop_session_control.go

```diff
@@ -0,0 +1,95 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const desktopSessionControlSummary = `allows setting up user processes for a desktop session`
+
+const desktopSessionBaseDeclarationSlots = `
+  desktop-session-control:
+    allow-installation:
+      slot-snap-type:
+        - app
+        - core
+    deny-auto-connection: true
+`
+
+const desktopSessionControlConnectedPlugAppArmor = `
+# Description: Can control the processes in desktop sessions, launch them, etc.
+
+/var/lib/snapd/desktop/applications/{,**} r,
+
+#include <abstractions/dbus-session-strict>
+#include <abstractions/dbus-strict>
+
+# Supporting session boot fully driven by user session systemd
+# and D-Bus activation
+dbus (send)
+    bus=session
+    path={/,/org/freedesktop/DBus}
+    interface=org.freedesktop.DBus
+    member=UpdateActivationEnvironment
+    peer=(label=unconfined),
+
+dbus (send)
+    bus=session
+    path=/org/freedesktop/systemd1
+    interface=org.freedesktop.DBus.Properties
+    member=Get{,All}
+    peer=(label=unconfined),
+
+dbus (send)
+    bus=session
+    path=/org/freedesktop/systemd1
+    interface=org.freedesktop.systemd1.Manager
+    member={SetEnvironment,UnsetEnvironment,UnsetAndSetEnvironment,ResetFailed,Reload,ListUnitFiles,ListUnitFilesByPatterns,StartUnit,StartTransientUnit}
+    peer=(label=unconfined),
+
+
+# Let the session get information about the current locale
+dbus (send)
+    bus=system
+    path=/org/freedesktop/locale1
+    interface=org.freedesktop.DBus.Properties
+    member=Get{,All},
+
+
+# KDE Plasma specific extension
+
+# Used by the KCrash handler
+@{PROC}/sys/kernel/core_pattern r,
+
+# So that KSplash disappears when appropriate
+dbus (send)
+    bus=session
+    path=/KSplash
+    interface=org.kde.KSplash
+    member=setStage,
+`
+
+func init() {
+	registerIface(&commonInterface{
+		name:                  "desktop-session-control",
+		summary:               desktopSessionControlSummary,
+		implicitOnCore:        true,
+		implicitOnClassic:     true,
```

Comment:
If this is intended for use by a snap acting as a desktop session, it might make sense not to provide the slot on classic.

If we find more use cases down the road, we can enable it then. That's easier than taking something away that was previously available.

Reply:
Indeed, this makes sense. I changed it now.